brave/brave-core
# feat(wallet): Visible Assets Modal V2

## Description 
- Updates the `Visible Assets` modal to the new `Wallet V2` designs
- Separates `My assets` and `Available assets` with a `Segmented Controller`.
- hashes (`#my-assets` or `#available-assets`) will be applied to the end of `route` depending on which option you click on the `Segmented Controller`. If no `hash` is present, it will default to showing `My assets`.

note:(We still need design mockups for the `Add Custom Asset` forms, I have created an issue for this here https://github.com/brave/brave-browser/issues/31504


Resolves <https://github.com/brave/brave-browser/issues/31389>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Open the `Visible Assets` modal on the `Portfolio` screen
2. By default `My assets` should be selected
3. Test `hiding` an asset and click `Save changes`, make sure it works
4. Reopen the `Visible Assets` modal and go to `Available assets` add a new asset and click `Save Changes` make sure it works 
5. Reopen the `Visible Assets` modal and go to the `Add custom asset` page,
6. Add a custom assets and make sure it works.

https://github.com/brave/brave-core/assets/40611140/ae07b99d-b1e6-4b5e-a0e3-ef1b9176da4b
